src/main/java/io/cdap/delta/bigquery/BigQueryAssessor.java

```diff
@@ -53,10 +62,58 @@ public TableAssessment assess(StandardizedTableDetail tableDetail) {
       }
     }
     List<Problem> problems = new ArrayList<>();
+    String dbName = tableDetail.getDatabase();
+    String tableName = tableDetail.getTable();
     if (tableDetail.getPrimaryKey().isEmpty()) {
-      problems.add(new Problem("Missing Primary Key", "Tables must have a primary key in order to be replicated.",
-                               "Please alter the table to use a primary key, or select a different table", ""));
+      problems.add(new Problem("Missing Primary Key",
+                               String.format("Table '%s' in database '%s' must have a primary key in order to be " +
+                                               "replicated", dbName, tableName),
+                               "Please alter the table to use a primary key, or select a different table",
+                               "Not able to replicate this table in BigQuery side"));
+    }
+
+    if (dbName.length() > MAX_LENGTH) {
+      problems.add(new Problem("Dataset Name Exceeding Max Length",
+                               String.format("Dataset name '%s' exceeds the max length %d defined in BigQuery",
+                                             dbName, MAX_LENGTH),
+                               "It's better to alter the database to make the name shorter",
```

Comment:
Verify that multiple databases will not be shortened to the same BigQuery dataset name.